### PR TITLE
Scan: Add loading state to AccountIndexPage

### LINKED ISSUE #1565: Implement Loading state on Account Index Page
Implement as design

### LINKED ISSUE #2538: Guanyu revamp - Account Index Page
Note:
Please merge #2363 before this or combine it to this PR.

### PR DESCRIPTION
**Pull request by pzshine**
Fixed: #1565
Fixed: #2538 

## Implementation details

 - Added Loading State to index page including the tab detail
 - `AccountIndexDelegations`, `AccountIndexUnbonding` and AccountIndexRedelegate
 - The loading state has been implemented on both desktop and mobile screens

-----

 - Adjusted the Account Index layout to the new style including these tabs below:
   - Transactions
   - Delegations
   - Unbonding
   - Redelegate
 - Fixed the spacing issue on Txtable
 - Fixed the spacing issue on `Msg.re`
 - Updated the no data image on each tab.

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Comment by pzshine** [comment]
Note:

The image needs to be changed on each tab, now I'm using the `noTransaction` image.

image

File: scan/src/components/account/AccountIndexRedelegate.re [hunk 1]
```diff
@@ -62,122 +114,172 @@ let renderBody =
...
+        ++ (completionTime |> MomentRe.Moment.toISOString)
+        ++ (amount |> Coin.getBandAmountFromCoin |> Js.Float.toString)
+        ++ (reserveIndex |> string_of_int)
+      }
+      idx={
+        (srcAddress |> Address.toBech32)
+        ++ (dstAddress |> Address.toBech32)
+        ++ (completionTime |> MomentRe.Moment.toISOString)
+        ++ (amount |> Coin.getBandAmountFromCoin |> Js.Float.toString)
+        ++ (reserveIndex |> string_of_int)
```
Comment (evilpeach) [inline scan/src/components/account/AccountIndexRedelegate.re]: It's duplicated. Let's make it a variable.

File: scan/src/components/account/AccountIndexUnbonding.re [hunk 2]
```diff
@@ -55,104 +82,148 @@ let renderBody = (unbondingEntry: UnbondingSub.unbonding_list_t) => {
...
+      ]
+      key={
+        (operatorAddress |> Address.toBech32)
+        ++ (completionTime |> MomentRe.Moment.toISOString)
+        ++ (reserveIndex |> string_of_int)
+      }
+      idx={
+        (operatorAddress |> Address.toBech32)
+        ++ (completionTime |> MomentRe.Moment.toISOString)
+        ++ (reserveIndex |> string_of_int)
```
Comment (evilpeach) [inline scan/src/components/account/AccountIndexUnbonding.re]: Make a variable here as well.

File: scan/src/pages/AccountIndexPage.re [hunk 3]
```diff
@@ -224,115 +218,163 @@ let make = (~address, ~hashtag: Route.account_tab_t) =>
...
-        let openSendModal = () =>
-          dispatchModal(OpenModal(SubmitTx(SubmitMsg.Send(Some(address)))));
-        if (sender == address) {
-          Window.confirm("Are you sure you want to send tokens to yourself?")
-            ? openSendModal() : ();
-        } else {
-          openSendModal();
-        };
-      | None => dispatchModal(OpenModal(Connect(tracking.chainID)))
+    totalBalance;
```
Comment (evilpeach) [inline scan/src/pages/AccountIndexPage.re]: It's no need to declare a variable. Just return `availableBalance +. .... `

File: scan/src/components/account/AccountIndexDelegations.re [hunk 4]
```diff
@@ -6,131 +6,210 @@ module Styles = {
...
+        ++ (amount |> Coin.getBandAmountFromCoin |> Js.Float.toString)
+        ++ (reward |> Coin.getBandAmountFromCoin |> Js.Float.toString)
+        ++ (reserveIndex |> string_of_int)
+      }
+      idx={
+        (operatorAddress |> Address.toHex)
+        ++ (amount |> Coin.getBandAmountFromCoin |> Js.Float.toString)
+        ++ (reward |> Coin.getBandAmountFromCoin |> Js.Float.toString)
+        ++ (reserveIndex |> string_of_int)
+      }
```
Comment (evilpeach) [inline scan/src/components/account/AccountIndexDelegations.re]: Make a variable.

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Comment by evilpeach** [comment]
> Note:
> 
> The image needs to be changed on each tab, now I'm using the `noTransaction` image.
> 
> image

Okay, let's ask a designer for a new image.

File: scan/src/components/TxsTable.re [hunk 5]
```diff
@@ -108,10 +83,10 @@ let renderBodyMobile =
     let msgTransform = messages->Belt_List.map(msgTransform);
     <MobileCard
       values=InfoMobileCard.[
-        ("TX HASH", TxHash(txHash, Media.isSmallMobile() ? 170 : 200)),
-        ("BLOCK", Height(blockHeight)),
-        ("GAS FEE\n(BAND)", Coin({value: gasFee, hasDenom: false})),
-        ("ACTIONS", Messages(txHash, msgTransform, success, errMsg)),
+        ("Tx Hash", TxHash(txHash, Media.isSmallMobile() ? 170 : 200)),
```
Comment (evilpeach) [inline scan/src/components/TxsTable.re]: TX Hash

File: scan/src/components/TxsTable.re [hunk 6]
```diff
@@ -120,10 +95,11 @@ let renderBodyMobile =
   | _ =>
     <MobileCard
       values=InfoMobileCard.[
-        ("TX HASH", Loading(Media.isSmallMobile() ? 170 : 200)),
-        ("BLOCK", Loading(70)),
+        ("Tx Hash", Loading(Media.isSmallMobile() ? 170 : 200)),
```
Comment (evilpeach) [inline scan/src/components/TxsTable.re]: TX Hash

File: scan/src/components/account/AccountIndexDelegations.re [hunk 7]
```diff
@@ -1,135 +1,210 @@
...
-           })
-         ->React.array}
-        <Pagination currentPage=page pageCount onPageChange={newPage => setPage(_ => newPage)} />
-      </>
-    </div>
-    |> Sub.resolve;
-  }
-  |> Sub.default(_, React.null);
+  <div className=Styles.tableWrapper>
+    <>
```
Comment (evilpeach) [inline scan/src/components/account/AccountIndexDelegations.re]: remove?

File: scan/src/components/account/AccountIndexRedelegate.re [hunk 8]
```diff
@@ -1,182 +1,271 @@
...
-                   </div>
-                 </Col>
-                 <Col> <HSpacing size=Spacing.lg /> </Col>
-               </Row>
-             </THead>}
-        {redelegateList
-         ->Belt.Array.map(redelegateEntry =>
-             isMobile ? renderBodyMobile(redelegateEntry) : renderBody(redelegateEntry)
+  <div className=Styles.tableWrapper>
+    <>
```
Comment (evilpeach) [inline scan/src/components/account/AccountIndexRedelegate.re]: remove?

File: scan/src/components/account/AccountIndexUnbonding.re [hunk 9]
```diff
@@ -1,157 +1,223 @@
...
-                   </div>
-                 </Col>
-                 <Col> <HSpacing size=Spacing.lg /> </Col>
-               </Row>
-             </THead>}
-        {unbondingList
-         ->Belt.Array.map(unbondingEntry =>
-             isMobile ? renderBodyMobile(unbondingEntry) : renderBody(unbondingEntry)
+  <div className=Styles.tableWrapper>
+    <>
```
Comment (evilpeach) [inline scan/src/components/account/AccountIndexUnbonding.re]: remove?

File: scan/src/pages/AccountIndexPage.re [hunk 10]
```diff
@@ -190,213 +178,240 @@ let totalBalanceRender = (isMobile, rawTitle, amount, symbol) => {
...
-          openSendModal();
-        };
-      | None => dispatchModal(OpenModal(Connect(tracking.chainID)))
+    availableBalance +. balanceAtStakeAmount +. rewardAmount +. unbondingAmount +. commissionAmount;
+  };
+  let send = chainID => {
+    switch (accountOpt) {
+    | Some({address: sender}) =>
+      let openSendModal = () =>
+        dispatchModal(OpenModal(SubmitTx(SubmitMsg.Send(Some(address)))));
```
Comment (evilpeach) [inline scan/src/pages/AccountIndexPage.re]: I think we should use this for easy understanding.
```
Some(Address) |> SubmitMsg.Send |> SubmitTx |> OpenModal |> dispatchModal
```

File: scan/src/pages/TxHomePage.re [hunk 11]
```diff
@@ -26,39 +6,54 @@ let make = () => {
...
-      </Row>
-      <VSpacing size=Spacing.xl />
+        </Col.Grid>
+      </Row.Grid>
+      {isMobile
+         ? React.null
+         : <THead.Grid>
+             <Row.Grid alignItems=Row.Center>
+               <Col.Grid col=Col.Two>
+                 <Text block=true value="Tx Hash" weight=Text.Semibold color=Colors.gray7 />
```
Comment (evilpeach) [inline scan/src/pages/TxHomePage.re]: TX Hash

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/components/account/AccountIndexDelegations.re [hunk 12]
```diff
@@ -1,135 +1,210 @@
...
+               <Heading
+                 size=Heading.H4
+                 value="No Delegation"
+                 align=Heading.Center
+                 weight=Heading.Regular
+                 color=Colors.bandBlue
+               />
+             </div>
+       | _ =>
+         Belt_Array.make(1, ApolloHooks.Subscription.NoData)
```
Comment (evilpeach) [inline scan/src/components/account/AccountIndexDelegations.re]: Let's add more to 10 rows.

File: scan/src/components/account/AccountIndexRedelegate.re [hunk 13]
```diff
@@ -1,182 +1,271 @@
...
+               <Heading
+                 size=Heading.H4
+                 value="No redelegation"
+                 align=Heading.Center
+                 weight=Heading.Regular
+                 color=Colors.bandBlue
+               />
+             </div>
+       | _ =>
+         Belt_Array.make(1, ApolloHooks.Subscription.NoData)
```
Comment (evilpeach) [inline scan/src/components/account/AccountIndexRedelegate.re]: here as well

File: scan/src/components/account/AccountIndexUnbonding.re [hunk 14]
```diff
@@ -1,157 +1,223 @@
...
+               <Heading
+                 size=Heading.H4
+                 value="No Unbonding"
+                 align=Heading.Center
+                 weight=Heading.Regular
+                 color=Colors.bandBlue
+               />
+             </div>
+       | _ =>
+         Belt_Array.make(1, ApolloHooks.Subscription.NoData)
```
Comment (evilpeach) [inline scan/src/components/account/AccountIndexUnbonding.re]: here as well

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — CHANGES_REQUESTED** [review]
Please check this
image
[Link](https://d3n-scan-pr-2363.onrender.com/account/band140lp4vgc0lvuvwef8pg4uqu9gg0vmv4ggn68jg)

File: scan/src/components/TxMessages.re [hunk 15]
```diff
@@ -3,10 +3,10 @@ module Styles = {
   let msgContainer = overflowed =>
     style([
       position(`relative),
-      height(overflowed ? `px(60) : `auto),
+      height(overflowed ? `px(45) : `auto),
       overflow(overflowed ? `hidden : `visible),
       selector("> div + div", [marginTop(`px(10))]),
-      Media.mobile([height(overflowed ? `px(50) : `auto)]),
+      // Media.mobile([height(overflowed ? `px(50) : `auto)]),
```
Comment (evilpeach) [inline scan/src/components/TxMessages.re]: remove

File: scan/src/components/account/AccountIndexUnbonding.re (hunk 9 above)
Comment (evilpeach) [inline scan/src/components/account/AccountIndexUnbonding.re]: remove

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**